ckan/lib/plugins.py

```diff
@@ -490,10 +491,25 @@ def bulk_process_template(self) -> str:
     def group_form(self) -> str:
         return 'group/new_group_form.html'
 
+    def create_group_schema(self) -> Schema:
+        return schema.default_create_group_schema()
+
+    def update_group_schema(self) -> Schema:
+        return schema.default_update_group_schema()
+
+    def show_group_schema(self) -> Schema:
+        return schema.default_show_group_schema()
+
+    # Deprecated schema methods, to be removed in CKAN 2.12
+
+    @deprecated(
+        "Use either `create_group_schema()` or `update_group_schema()`",
+        since="2.11.0"
+    )
     def form_to_db_schema_options(self,
                                   options: dict[str, Any]) -> dict[str, Any]:
-        ''' This allows us to select different schemas for different
-        purpose eg via the web interface or via the api or creation vs
+        ''' [Deprecated] This allows us to select different schemas for
```

Comment:
Not only deprecated, these won't be called at all any more even as a fall-back, right?

Reply:
(looks like ckanext-scheming uses the `validate` method of `IGroupForm` it won't be affected by the change)

Reply:
Correct. I thought of leaving them for a version in case some plugin was using them but that's seems unlikely and maybe we can just remove them?

Reply:
The polite thing would be to have a version released with the methods deprecated and working as a fall-back then remove them next release.

Marking them deprecated but actually broken doesn't make sense. Better to remove them and raise an error when a plugin tries to use them.

Reply:
Just to make sure @wardi , you are suggesting that in the actions, if an `IGroupForm` does not have one of the new methods (eg `IGroupForm.create_group_schema()` it should try to call the old equivalent, eg `IGroupForm.form_to_db_schema()` right?

Reply:
yes, with the deprecation warning and remove it the next release

Reply:
@wardi see https://github.com/ckan/ckan/pull/8069/commits/bd720fd0c959e56eb237c6a787eee75d7a09ca73